tests/test_aws/test_ecs.py

```diff
@@ -0,0 +1,122 @@
+    assert (
+        ecs_client.task_definition
+        == "arn:aws:ecs:us-east-1:123456789012:task-definition/mock-sapinvoices-ecs-test:1"
+    )
+    assert ecs_client.task_family_revision == "mock-sapinvoices-ecs-test:1"
+
+
+def test_ecs_client_task_family_property_success(ecs_client):
+    assert (
+        ecs_client.task_definition
+        == "arn:aws:ecs:us-east-1:123456789012:task-definition/mock-sapinvoices-ecs-test:1"
+    )
+    assert ecs_client.task_family == "mock-sapinvoices-ecs-test"
+
+
+def test_ecs_client_execute_review_run_success(
+    ecs_client, ecs_client_execute_run_details_success, caplog
+):
+    assert ecs_client.execute_review_run() == {
+        "task_run_arn": "arn:aws:ecs:us-east-1:123456789012:task/mock-sapinvoices-ecs-test/abc123",
+        "status": "PROVISIONING",
+        "created_at": "2024-07-02T10:16:43.077000-04:00",
+        "stopped_at": None,
+    }
+    assert "Executing ECS task for 'Review Run'" in caplog.text
+
+
+def test_ecs_client_execute_final_run_success(
+    ecs_client, ecs_client_execute_run_details_success, caplog
+):
+    assert ecs_client.execute_final_run() == {
+        "task_run_arn": "arn:aws:ecs:us-east-1:123456789012:task/mock-sapinvoices-ecs-test/abc123",
+        "status": "PROVISIONING",
+        "created_at": "2024-07-02T10:16:43.077000-04:00",
+        "stopped_at": None,
+    }
+    assert "Executing ECS task for 'Final Run'" in caplog.text
+
+
+def test_ecs_client_run_task_if_run_type_is_invalid(ecs_client):
+    with pytest.raises(
+        ValueError, match="Cannot run task for unrecognized run_type='invalid'"
+    ):
+        ecs_client.run(run_type="invalid")
+
+
+def test_ecs_client_monitor_task_success(
+    ecs_client, mock_ecs_task_state_transitions, caplog
+):
+    """Mock iterative ECSClient.monitor_task method.
+
+    ECS task state transitions are mocked. Passing this test verifies that
+    ECSClient.monitor_task will repeatedly make calls to boto3.ECS.Client.describe_tasks
+    until the task is completed (i.e., status="STOPPED") by checking that each
+    transition is captured in the logs.
+
+    Note: The captured state transitions comprise of the last four stages of the
+          ECS task lifecycle.
+          See https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task-lifecycle-explanation.html
+    """
+    task_arn = mock_ecs_task_state_transitions
+    ecs_client.monitor_task(task_arn)
+    for status in ("DEACTIVATING", "STOPPING", "DEPROVISIONING", "STOPPED"):
+        assert status in caplog.text
+
+    assert "Task run has completed." in caplog.text
+
+
+def test_ecs_client_monitor_task_raise_timeout_error(
+    ecs_client, mock_ecs_task_state_transitions, caplog
+):
+
+    task_arn = mock_ecs_task_state_transitions
+
+    with pytest.raises(
+        ECSTaskRuntimeExceededTimeoutError,
+        match="Task runtime exceeded set timeout of 2 seconds.",
+    ):
+        ecs_client.monitor_task(task_arn, timeout=2)
```

Comment:
Nice test!  With your 5 second loop, this is a really nice way to exercise this timeout logic.